iwyu_tool.py

```diff
@@ -111,35 +111,44 @@ def find_include_what_you_use():
 IWYU_EXECUTABLE = find_include_what_you_use()
 
 
-def run_iwyu(dbentry, extra_args, verbose):
-    """ Transform dbentry to an IWYU command, and run it. """
-    cwd = dbentry['directory']
-
-    if 'arguments' in dbentry:
-        # arguments is a command-line in list form.
-        command = dbentry['arguments']
-    elif 'command' in dbentry:
-        # command is a command-line in string form, split to list.
-        command = shlex.split(dbentry['command'])
-    else:
-        raise ValueError('Invalid compilation database entry: %s' % dbentry)
-
-    if not IWYU_EXECUTABLE:
-        raise ValueError('No include-what-you-use found on PATH')
-
-    # Rewrite the compile command for IWYU
-    compile_command, compile_args = command[0], command[1:]
-    if compile_command.endswith('cl.exe'):
-        # If the compiler name is cl.exe, let IWYU be cl-compatible.
-        extra_args = ['--driver-mode=cl'] + extra_args
-
-    command = [IWYU_EXECUTABLE] + extra_args + compile_args
-
+class Invocation(object):
+    """ Holds arguments of an IWYU invocation. """
+    def __init__(self, command, cwd):
+        self.command = command
+        self.cwd = cwd
+
+    def __str__(self):
+        return ' '.join(self.command)
+
+    @staticmethod
+    def parse_compile_command(entry, extra_args):
```

Comment:
Misleading name. Maybe, `create_from_compile_command`?

Reply:
Or simply, `from_compile_command`?

Reply:
I have a tendency to call everything that translates from something less desirable to something more desirable "parse" :) But `from_` is shorter than `parse_`, so I'll take that -- good suggestion, thanks!

Reply:
Updated.